test/integration/api/services_controller_test.rb

```diff
@@ -48,36 +51,96 @@ class Api::ServicesControllerTest < ActionDispatch::IntegrationTest
+    end
+    assert Service.last.act_as_product
+  end
+
+  test 'should not mark that service will act as product if feature is not enabled' do
+    Account.any_instance.stubs(:provider_can_use?).with(:api_as_product).returns(false).at_least_once
+
+    assert_change of: -> { Service.count } do
+      post admin_services_path, service: {
+        system_name: 'my_new_product',
+        name: 'My new Product',
+        description: 'This will act as product',
+        act_as_product: true
+      }
+    end
+    refute Service.last.act_as_product
+  end
+
+  test 'should create a new Backend API if none was selected' do
+    Account.any_instance.stubs(:provider_can_use?).with(:api_as_product).returns(true).at_least_once
+
+    assert_change of: -> { BackendApi } do
+      post admin_services_path, service: {
+        system_name: 'my_new_product',
+        name: 'My new Product',
+        description: 'This will act as product',
+        act_as_product: true
+      }
+    end
+
+
+  end
+
+  test 'should reuse the same Backend API if it was selected' do
+    Account.any_instance.stubs(:provider_can_use?).with(:api_as_product).returns(true).at_least_once
+    backend_api = FactoryBot.create(:backend_api, account: @provider)
+
+    assert_change of: -> { BackendApi }, by: 0 do
+      post admin_services_path, service: {
+        system_name: 'my_new_product',
+        name: 'My new Product',
+        description: 'This will act as product',
+        act_as_product: true,
+        backend_api: backend_api.id
+      }
+    end
+
+    assert_equal backend_api, Service.last.backend_api
+  end
+
   private
 
   def update_params
     @update_params ||= { service:
-                             { intentions_required: '0',
-                               buyers_manage_apps: '0',
-                               referrer_filters_required: '1',
-                               custom_keys_enabled: '1',
-                               buyer_can_select_plan: '1',
-                               buyer_plan_change_permission: 'direct',
-                               notification_settings:
-                                   { web_provider: ['', '50', '100', '300'],
-                                     email_provider: ['', '50', '100', '150'],
-                                     web_buyer: ['', '50', '100', '150'],
-                                     email_buyer: ['', '50', '100', '300']
-                                   }
-                             }
+      { intentions_required: '0',
+        buyers_manage_apps: '0',
+        referrer_filters_required: '1',
+        custom_keys_enabled: '1',
+        buyer_can_select_plan: '1',
+        buyer_plan_change_permission: 'direct',
+        notification_settings:
+        { web_provider: ['', '50', '100', '300'],
+          email_provider: ['', '50', '100', '150'],
+          web_buyer: ['', '50', '100', '150'],
+          email_buyer: ['', '50', '100', '300']
+        }
+      }
     }
   end
 
-end
+  end
```

Comment:
Wrong indentation I think:
```suggestion
end
```